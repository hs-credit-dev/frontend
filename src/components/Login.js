import React, { useState } from 'react';
import axios from 'axios';
import jwt_decode from 'jwt-decode';
import { Redirect } from 'react-router-dom';
import setAuthToken from '../utilities/setAuthToken'


const REACT_APP_DATABASE_URL = process.env.REACT_APP_DATABASE_URL;

const Login = (props) => {
    const [email, setEmail] = useState('');
    const [password, setPassword] = useState('');

 

    const handleEmail = (e) => {
        setEmail(e.target.value);
    }

    const handlePassword = (e) => {
        setPassword(e.target.value);
    }

    const handleSubmit = (e) => {
        console.log(`>>>> inside handleSubmit func`)
        e.preventDefault()
        const userData = {email,password}

        axios.post(`${REACT_APP_DATABASE_URL}/users/login`, userData)
        .then(response => {
            console.log(`>>>>inside handleSubmit reponse block`)

            console.log(response)
            const { token } = response.data
            //save token to localStorage
            localStorage.setItem('jwtToken', token)
            //set token to header
            setAuthToken(token)
            //decode token to get user data
            const decoded = jwt_decode(token)
            props.nowCurrentUser(decoded)
        })
        .catch(error => {
            console.log(`>>>> error logging in ${error}`)
            alert('incorrect email or password! please try again')
        })
    }

    if (props.user) return <Redirect to="/profile" user={props.user}/> // double check

    return (
       
        <div className="login-container">
            <div>
                <div>
                <h1 className="login">Login</h1>
                    <form className="login-form"onSubmit={handleSubmit}>
                        <div className="form-group">
                            <label htmlFor="email">Email  </label>
                            <input type="text" name="email" value={email} onChange={handleEmail} className="form-control"/>
                        </div>
                        <div className="form-group">
                            <label htmlFor="password">Password  </label>
                            <input type="password" name="password" value={password} onChange={handlePassword} className="form-control"/>
                        </div>
                        <button type="submit" className="form-group">Submit</button>
                    </form>
                </div>
            </div>
        </div>

    )
}


export default Login;
