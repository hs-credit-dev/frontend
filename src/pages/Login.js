// imports
import React, { useState } from 'react';
import axios from 'axios';
// import jwt_decode from 'jwt-decode';
import { Redirect } from 'react-router-dom';
// import setAuthToken from '../utilities/setAuthToken';
import logo from '../assets/images/hscLogo21.png';
import Footer from '../components/Footers/AccountCreationFooter';

import '../styles/login.styles.css'

// storing database url in environment
const { REACT_APP_DATABASE_URL } = process.env

const Login = (props) => {

    // state variables
    const [email, setEmail] = useState('');
    const [password, setPassword] = useState('');
    const [redirect, setRedirect] = useState(false)

    // set email based on event
    const handleEmail = (e) => {
        setEmail(e.target.value);
    }

    // set password based on event
    const handlePassword = (e) => {
        setPassword(e.target.value);
    }

    // handle submit function
    const handleSubmit = async (e) => {
        console.log(`>>>> inside handleSubmit func`)
        e.preventDefault()
        const userData = { email, password }

        try 
        {
            const response = await axios.post(`${REACT_APP_DATABASE_URL}/users/login`, userData)
            console.log(`>>>>inside handleSubmit reponse block`)
            const token = await response.data
            await localStorage.setItem('jwtToken', token)
            // setAuthToken(token)
            // const decoded = jwt_decode(token)
            // props.nowCurrentUser(decoded)
            setRedirect(true) 
        }
        catch(error)
        {
            console.log(`>>>> error in handleSubmit func: ${error}`)
            alert('incorrect email or password! please try again')
        }

    }

    // if there is user logged in, redirect them home
    // if (props.user) return <Redirect to="/home" />

    return (
        <div>
            <div className="login-container">
                <div className="logo-title-banner">
                    <img src={logo} className="logo-image"/>
                    <a className="title">hs.credit</a>
                </div>

                <form className="login-form" onSubmit={handleSubmit}>
                    <input type="text" name="email" value={email} onChange={handleEmail} placeholder="Username" className="form-input" />
                    <input type="password" name="password" value={password} onChange={handlePassword} placeholder="Password" className="form-input"/>
                    <button type="submit" className="login-button">Log In</button>
                    <a className='forgot-password-link' href="">Forgot password?</a>
                </form>

                <button href="www.google.com" className="create-account-button">Create account</button>
            </div>
            <Footer />
        </div>

    )
}


export default Login;
