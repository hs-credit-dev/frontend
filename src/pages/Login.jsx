import { useState, useEffect } from "react";
import { useNavigate, NavLink } from "react-router-dom";
import { useAtom } from "jotai";

import { users } from "../api_fake";
import { userInSession } from "../App";

import Logo from "../components/Logo";
import Input from "../components/Input";
import Button from "../components/Button";
import SubmitButton from "../components/SubmitButton";

const Login = () => {
  const [user, setUser] = useAtom(userInSession);

  const [username, setUsername] = useState("");
  const [password, setPassword] = useState("");
  const [warning, setWarning] = useState("");

  const navigate = useNavigate();

  const handleSubmit = async (e) => {
    e.preventDefault();

    if (!username || !password) return;

    try {
      setWarning("");
      const res = await users.login(username, password);
      const user = res.data.data.user;
      if (!user) {
        setWarning("incorrect username or password");
        return;
      }
      setUser(user);
      navigate("/");
    } catch (err) { }
  };

  useEffect(() => {
    if (user) {
      return navigate('/dashboard');
    }
  }, [user, navigate]);

  return (
    <>
      <div className="flex flex-col align-middle my-auto h-4/5">
        <div className="flex flex-col gap-6 mt-auto">
          {/* <NavLink to="/"> */}
          <Logo className="mx-auto mb-4" />
          {/* </NavLink> */}
          <form className="flex flex-col space-y-6" onSubmit={handleSubmit}>
            <Input
              name="username"
              autoComplete="username"
              placeholder="Username"
              className="mx-auto text-center w-4/5 sm:w-full max-w-lg"
              onChange={(e) => {
                setWarning("");
                setUsername(e.target.value);
              }}
            />
            <Input
              type="password"
              autoComplete="current-password"
              name="password"
              placeholder="Password"
              className="mx-auto text-center w-4/5 sm:w-full max-w-lg"
              onChange={(e) => {
                setWarning("");
                setPassword(e.target.value);
              }}
            />
            <SubmitButton name="Log In" className="mx-auto" />
          </form>
          <a className="mx-auto font-bold text-blue-700" href="/">
            Forgot password?
          </a>
          <p className="mx-auto text-red-600 font-bold">{warning}</p>
        </div>
        <NavLink to="/signup" className="mx-auto mb-12 mt-auto">
          <Button>Create account</Button>
        </NavLink>
      </div>
    </>
  );
};

export default Login;
