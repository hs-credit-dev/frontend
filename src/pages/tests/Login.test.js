import renderer from 'react-test-renderer';
import Login from '../Login'


describe('Login', () => {

    it('matches snapshot', () => {
    const component = renderer.create(
      <Login />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});

