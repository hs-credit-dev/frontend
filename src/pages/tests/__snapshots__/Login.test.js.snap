// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login matches snapshot 1`] = `
<div
  className="login-container"
>
  <div>
    <div
      className="login-div"
    >
      <h1
        className="login"
      >
        Login
      </h1>
      <form
        className="login-form"
        onSubmit={[Function]}
      >
        <div
          className="form-group"
        >
          <input
            className="form-control"
            name="email"
            onChange={[Function]}
            type="text"
            value=""
          />
          <p
            className="input-text"
          >
            (email)
          </p>
        </div>
        <div
          className="form-group"
        >
          <input
            className="form-control"
            name="password"
            onChange={[Function]}
            type="password"
            value=""
          />
          <p
            className="input-text"
          >
            (password)
          </p>
        </div>
        <button
          className="form-group-button"
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </div>
</div>
`;
